app/src/main/cpp/skyline/loader/nso.h

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#pragma once
+
+#include <common.h>
+#include "loader.h"
+
+namespace skyline::loader {
+    /**
+     * @brief The NsoLoader class abstracts access to an NSO file through the Loader interface (https://switchbrew.org/wiki/NSO)
+     */
+    class NsoLoader : public Loader {
+      private:
+        union NsoFlags {
+            struct {
+                bool textCompressed : 1; //!< .text is compressed
+                bool roCompressed : 1; //!< .rodata is compressed
+                bool dataCompressed : 1; //!< .data is compressed
+                bool textHash : 1; //!< .text hash should be checked before loading
+                bool roHash : 1; //!< .rodata hash should be checked before loading
+                bool dataHash : 1; //!< .data hash should be checked before loading
+            };
+            u32 raw; //!< The raw value of the flags
+        };
+        static_assert(sizeof(NsoFlags) == 0x4);
+
+        /**
+         * @brief This holds a single data segment's offset, loading offset and size
+         */
+        struct NsoSegmentHeader {
+            u32 fileOffset; //!< The offset of the segment in the NSO
+            u32 memoryOffset; //!< The memory offset where the region should be loaded
+            u32 decompressedSize; //!< Size of the region after decompression
+        };
+        static_assert(sizeof(NsoSegmentHeader) == 0xc);
+
+        /**
+         * @brief This holds the header of an NSO file
```

Comment:
"This structure holds..." 😉

Reply:
There are about a million other comments that sat 'this holds'